pkg/api/pin_chunks.go

```diff
@@ -184,6 +187,129 @@ func (s *server) getPinnedChunk(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+type updatePinCounter struct {
+	PinCounter uint64 `json:"pinCounter"`
+}
+
+// updatePinnedChunkPinCounter allows changing the pin counter for the chunk.
+func (s *server) updatePinnedChunkPinCounter(w http.ResponseWriter, r *http.Request) {
+	addr, err := swarm.ParseHexAddress(mux.Vars(r)["address"])
+	if err != nil {
+		s.Logger.Debugf("update pin counter: parse chunk ddress: %v", err)
+		s.Logger.Errorf("update pin counter: parse address")
+		jsonhttp.NotFound(w, nil)
+		return
+	}
+
+	has, err := s.Storer.Has(r.Context(), addr)
+	if err != nil {
+		s.Logger.Debugf("update pin counter: localstore has: %v", err)
+		s.Logger.Errorf("update pin counter: store")
+		jsonhttp.NotFound(w, nil)
```

Comment:
This should be internal server error, as `!has` is not found.

Reply:
I copied existing `getPinnedChunk` function. Will change it there also.